fecgov/fec-cms
# [WIP] Sitemaps

## Summary (required)

- Resolves #5522 

We are transitioning away from search.gov i14Y drawer  to using sitemaps.
- Sitemaps:
     - One sitemap for Wagtail pages generated  by Wagtail CMS  named:  `/sitemap-wagtail.xml` 
     - One for non-cms-generated pages named `sitemap-static.xml` that is manually created/updated. Since we cannot calculate `lastmod dates` on these, we will add them to the URLs when applicable or email search.gov if we  want to  trigger a re-index. [See comment in search.gov requirements ticket. ](https://github.com/fecgov/fec-cms/issues/5501#issuecomment-1505989503)
- Roots.txt:
    - The sitemaps will be referenced by search.gov via the new `robots_prod.txt` only in production. 
     - The other,  `robots.txt` ,tells search engines to Disallow crawling in non-prod  envs (stage. dev, feature).
- The  sitemaps themselves do not need to be conditionally hidden at certain environments because  they always reference `www.fec.gov`, so they would not ever prompt an search engine to crawl stage, dev etc, subdomains. Also because they are not named `sitemap.xml`, which is what the search engine looks for.
- [ ] **We  need to try to setup a test environment for this in dev. For pollicy-guidance , maybe  just a use a subset of policy-guidance  aliased pages to test the isolation of those pages from global search** See t[his issue ](https://github.com/fecgov/fec-cms/issues/5595)
      - [ ] Turn off  password protection on dev.fec.gov home page (wich password protects all its chid wagtail pages)
      - [ ] change site settings in Wagtail admin ( Settings > Sites ) to `host: dev.fec.gov` , `port: 443`
- [ ] **TODO**: See [below comment](https://github.com/fecgov/fec-cms/pull/5689#issuecomment-1496892806) about changes to `wagtail_hooks` that references `search_indexing.py`
- [ ] **TODO** Delete the search.gov I14Y drawer. Check with search.gov support on the best strategy to do this. 
- [ ] **TODO** We can remove drawer handle  keys  from base.py and from cloud.gov user-provided env vars..
- [ ]  ** TODO** Create new sites for prod global search and prod policy-giudance search. Get the keys from the Activate (</>) tab and replace them in cloud.gov for all environments. They will replace `SEARCHGOV_API_ACCESS_KEY` and `SEARCHGOV_POLICY_GUIDANCE_KEY` . (FYI, they are also referenced in `base.py` and `search/views.py`)
- [ ] **TODO**  See [this comment](https://github.com/fecgov/fec-cms/pull/5735#issuecomment-1549903664) on deleting non-existent registration-and-reporting  pages that show up in sitemap: These pages have been removed from dev database, and still need to be removed from prod before launching site-map
- [ ] Turn  password protection back on for dev.fec.gov home page (which password protects all its chid wagtail pages)

**Related Issues:** 
Research sitemap generation -  https://github.com/fecgov/fec-cms/issues/5499
Document search.gov sitemap - requirements https://github.com/fecgov/fec-cms/issues/5501
Changes to Policy and Guidance search before transitioning to sitemaps for global search - https://github.com/fecgov/fec-cms/issues/5595)https://github.com/fecgov/fec-cms/issues/5595

### Required reviewers

(Include who is required to review prior to merge. For example: One designer and two front end developer reviews are required prior to merge)

## Impacted areas of the application

General components of the application that this PR will affect:

-  

## Screenshots

(Include a screenshot of the new/updated features in context (“in the wild”). If it is an interface change, include both before and after screenshots)

## Related PRs

Related PRs against other branches:

branch | PR
------ | ------
fix/other_pr | [link]()
feature/other_pr | [link]()

## How to test

-  Test  to `127.0.0.1:8000/robots.txt` with env set to `stage` and  `prod` . You  should see a `disallow` robots.txt in stage, and `sitemap listings` in prod. Also see the conditional statements in  `urls.py` to confirm what  you should see at different envs/URLS.
-  The  sitemaps are not hidden at certain environmemts because  they always reference `www.fec.gov`, so they would not ever prompt an SE to crawl stagge, dev etc, subdomains. Also because they are not named `sitemap.xml`, which is what the search engine looks for.
    -127.0.0.1:8000/robots.txt/sitemap-wagtail.xml
     -127.0.0.1:8000/robots.txt/sitemap-static.xml (this is still WIP)
- Run `pytest` . The  new tests confirm the content of the response at  `/robots.txt` given a certain env (i.e stage, prod)
